### PR TITLE
Orca: add pure tf2/torch test environment

### PR DESCRIPTION
**Pull request by hamham223**
### 1. Why the change?

Setup a clean orca test env that only has tf2 / torch. 

Related PR: https://github.com/intel-analytics/BigDL/pull/7463

### 2. Summary of the change 

+ `test/bigdl/orca/torch/`: rm unnecessary `import`
+ `dev/test`: two test scripts
+ `PR_validation.yml` and `action.yml`: triggers

### 3. How to test?
The following unit test should be completed ✅ 

- [ ] Orca-Python-Py37-Pytorch
- [ ] Orca-Python-Py37-Tf2
- [ ] Orca-Jep-Python-Py37-Spark2

### 4. TO-DO

1. Rename the test
2. Polish the test scripts

### CODE REVIEW

File: python/orca/dev/test/run-pytests-tf2.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_get_model
+python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_optional_model_creator_h5
+python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_optional_model_creator_savemodel
+# require uuid
+# python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_save_load_model_architecture
+python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_save_load_model_h5     
+python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_save_load_model_savemodel
+python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_tensorboard          
+# require pandas
+# python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_xshards_pandas_dataframe
```
Comment (hamham223) [inline python/orca/dev/test/run-pytests-tf2.sh:48]: If all the above test can pass in the unit test, later I will move `test_save_load_model_arhitecture` and `test_xshards_pandas_dataframe` to a new test class. And the test script can be simplified.. 😆

File: python/orca/test/bigdl/orca/torch/test_torch_estimator.py [hunk 2]
```diff
@@ -16,7 +16,6 @@
 
 import torch
 from torch import nn
-import torchvision
```
Comment (hamham223) [inline python/orca/test/bigdl/orca/torch/test_torch_estimator.py:19]: These `import`s are unused, thus removed.

File: python/orca/dev/test/run-pytests-pytorch.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+cd "`dirname $0`"
+cd ../..
+
+export PYSPARK_PYTHON=python
+export PYSPARK_DRIVER_PYTHON=python
+
+echo "Running Orca Pytorch Test"
+python -m pytest -v test/bigdl/orca/test_import.py::TestImport::test_pytorch_import
+python -m pytest -v test/bigdl/orca/learn/test_metrics.py
+python -m pytest -v test/bigdl/orca/torch/
```
Comment (hamham223) [inline python/orca/dev/test/run-pytests-pytorch.sh:33]: If `orca/torch` can pass here, it can be removed from the jep test

Comment (hamham223) [reply python/orca/dev/test/run-pytests-pytorch.sh:33]: https://github.com/intel-analytics/BigDL/actions/runs/4159708826/jobs/7197006821#step:6:7109 
FAILED, some test need jep, will remove this line.

File: python/orca/dev/test/run-pytests-tf2.sh [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_dataframe    
+python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_dataframe_different_train_val
+python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_dataframe_shard_size 
+python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_dataframe_with_empty_partition
+python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_get_model
+python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_optional_model_creator_h5
+python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_optional_model_creator_savemodel
+# require uuid
+# python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_save_load_model_architecture
+python -m pytest -v test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py::TestTFEstimator::test_save_load_model_h5
```
Comment (hamham223) [inline python/orca/dev/test/run-pytests-tf2.sh:44]: FAILED due to random error.
https://github.com/intel-analytics/BigDL/actions/runs/4159708826/jobs/7197007197

File: python/orca/dev/test/run-pytests-pytorch.sh [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+cd "`dirname $0`"
+cd ../..
+
+export PYSPARK_PYTHON=python
+export PYSPARK_DRIVER_PYTHON=python
+
+echo "Running Orca Pytorch Test"
+python -m pytest -v test/bigdl/orca/test_import.py::TestImport::test_pytorch_import
+python -m pytest -v test/bigdl/orca/learn/test_metrics.py
```
Comment (hamham223) [inline python/orca/dev/test/run-pytests-pytorch.sh:32]: All `test_metrics.py` passed

File: .github/actions/orca/orca-python-py37-spark3-pytorch/action.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        export SPARK_LOCAL_HOSTNAME=localhost
+        
+        bash python/dev/release_default_linux_spark313.sh default false false false -Ddata-store-url=$HTTP_URI -U
+        
+        # install dllib
+        pip install -i https://pypi.org/simple python/dllib/src/dist/bigdl_dllib*-py3-none-manylinux1_x86_64.whl
+        # install orca
+        pip install -i https://pypi.org/simple python/orca/src/dist/bigdl_orca*-py3-none-manylinux1_x86_64.whl
+     
+        pip install torch==1.7.1
```
Comment (hamham223) [inline .github/actions/orca/orca-python-py37-spark3-pytorch/action.yml:19]: #### TODO:
add pip mirror

File: python/orca/dev/test/run-pytests-tf2.sh (hunk 1 above)
Comment (hamham223) [reply python/orca/dev/test/run-pytests-tf2.sh]: updated.

File: python/orca/dev/test/run-pytests-pytorch.sh (hunk 5 above)
Comment (hkvision) [reply python/orca/dev/test/run-pytests-pytorch.sh]: don't need to run test_metrics here.

File: python/orca/test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py [hunk 7]
```diff
@@ -27,9 +27,6 @@
 from bigdl.orca import OrcaContext
 
 
-resource_path = os.path.join(
```
Comment (hamham223) [inline python/orca/test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py:30]: unused, thus removed

File: .github/actions/orca/orca-basic-tf2-py37-spark3-action/action.yml [hunk 8]
```diff
@@ -0,0 +1,26 @@
+name: 'Run Orca Python Py37 Pytorch'
+description: 'Run Orca Python Pytorch'
```
Comment (hkvision) [inline .github/actions/orca/orca-basic-tf2-py37-spark3-action/action.yml:2]: wrong

Comment (hamham223) [reply .github/actions/orca/orca-basic-tf2-py37-spark3-action/action.yml]: fixed, btw this will not be displayed 😶‍🌫️

**Pull request merged**